### ISSUE TITLE
The provider doesn't rebuild an image when there are changes to a copied folder

### ISSUE THREAD

**Issue by voroninman** [kreuzwerker/terraform-provider-docker#302]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and docker Provider) Version

```
# terraform -v
Terraform v1.0.9
on darwin_amd64
+ provider registry.terraform.io/kreuzwerker/docker v2.15.0
```

### Affected Resource(s)

* `docker_image`
* `docker_registry_image`

### Terraform Configuration Files

`main.tf`:
```hcl
terraform {
  required_providers {
    docker = {
      source = "kreuzwerker/docker"
    }
  }
}

resource "docker_image" "this" {
  name = "terraform-docker-playground"
  build {
    path = "${path.module}/docker"
  }
}
```
`docker/Dockerfile`:
```Dockerfile
FROM python:3
COPY app /app
```

### Expected Behaviour

Terraform rebuilds the image when the content of `app` folder changes.

### Actual Behaviour

Terraform produces an empty plan.

### Steps to Reproduce

1. `terraform apply`
2. `echo "new data" >> docker/app/data.txt`
3. `terraform apply`

**Comment by Makeshift** [comment]
A slightly uglier workaround that doesn't utilise archives:

```hcl
locals {
  context = "${path.cwd}/containers/prometheus"
}

resource "docker_image" "promethus" {
  name = "${module.prometheus.ecr_repo}:latest"
  build {
    path = local.context
    tag  = [sha1(join("", [for file in fileset(local.context, "*") : filesha1("${local.context}/${file}")])), "latest"]
  }
}
```

**Comment by Makeshift** [comment]
Automatically closing issues is a pet peeve of mine.

**Comment by Junkern** [comment]
Reopening. I will have to check whether this is still a problem.

**Comment by Junkern** [comment]
I have implemented a `triggers` attribute (with `v2.20.1`), which you can for example use to rebuild the image whenever the contents of a folder change. Please see https://registry.terraform.io/providers/kreuzwerker/docker/latest/docs/resources/image#build for an example.
